packages/frontend/src/utils/staking.js

```diff
@@ -93,3 +96,40 @@ export function shuffle(sourceArray) {
     }
     return sourceArray;
 }
+
+const SECONDS_IN_YEAR = 3600 * 24 * 365;
+const NEAR_TOKEN_ID = 'wrap.near';
+
+export const calculateAPY = async (validator) => {
+    // Handle if there are no farms:
+    const activeFarms = validator?.poolSummary?.farms;
+    if (!activeFarms) return 0;
+
+    try {
+        const tokenPrices = fetchTokenPrices();
+
+        // TODO: check handle multiple farms, if some of token prices are unavailable.
+        if (activeFarms.some(farm => !+tokenPrices[farm.token_id]?.price)) return 0;
+
+        const totalStakedBalance = validator?.poolSummary.total_staked_balance;
+
+        const summaryAPY = activeFarms.reduce((acc, farm) => {
+            const tokenPriceInUSD = +tokenPrices[farm.token_id].price;
+            const nearPriceInUSD = +tokenPrices[NEAR_TOKEN_ID].price;
+
+            const rewardsPerSecond = farm.amount / (farm.end_date - farm.start_date) * 1e9;
+            const rewardsPerSecondInUSD = rewardsPerSecond * tokenPriceInUSD;
```

Comment:
All calculation should be with big numbers

Reply:
Check please the correctness of calculations